File: common/src/main/java/net/ps/weaponinfusion/WeaponInfusion.java
package net.ps.weaponinfusion;

import net.minecraft.core.registries.Registries;
import net.minecraft.resources.Identifier;
import net.blay09.mods.balm.core.BalmRegistrars;
import net.ps.weaponinfusion.data.ModDataComponents;
import net.ps.weaponinfusion.event.ModEvents;
import net.ps.weaponinfusion.potion.ModPotions;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import net.ps.weaponinfusion.item.ModItems;

public class WeaponInfusion {
    private WeaponInfusion() {
        /* This utility class should not be instantiated */
    }

    public static final Logger logger = LoggerFactory.getLogger(WeaponInfusion.class);

    public static final String MOD_ID = "weaponinfusion";

    public static Identifier id(String path) {
        return Identifier.fromNamespaceAndPath(MOD_ID, path);
    }

    public static void initialize(BalmRegistrars registrars) {
        registrars.items(ModItems::initialize);
        registrars.registrar(Registries.POTION, ModPotions::initialize);
        registrars.creativeModeTabs(ModItems::initialize);

        registrars.dataComponentTypes(ModDataComponents::initialize);

        ModEvents.initialize();
    }
}

File: common/src/main/java/net/ps/weaponinfusion/WeaponInfusion.kt
package net.ps.weaponinfusion

import net.blay09.mods.balm.core.BalmRegistrars
import net.minecraft.core.registries.Registries
import net.minecraft.resources.Identifier
import net.ps.weaponinfusion.data.ModDataComponents
import net.ps.weaponinfusion.event.ModEvents
import net.ps.weaponinfusion.item.ModItems
import net.ps.weaponinfusion.potion.ModPotions
import net.ps.weaponinfusion.sound.ModSoundEvents
import org.slf4j.Logger
import org.slf4j.LoggerFactory

object WeaponInfusion {
    val logger: Logger = LoggerFactory.getLogger(WeaponInfusion::class.java)

    const val MOD_ID: String = "weaponinfusion"

    fun id(path: String): Identifier {
        return Identifier.fromNamespaceAndPath(MOD_ID, path)
    }

    fun initialize(registrars: BalmRegistrars) {
        registrars.items(ModItems::initialize)
        registrars.creativeModeTabs(ModItems::initialize)

        registrars.dataComponentTypes(ModDataComponents::initialize)

        registrars.registrar(Registries.POTION, ModPotions::initialize)
        registrars.registrar(Registries.SOUND_EVENT, ModSoundEvents::initialize)

        ModEvents.initialize()
    }
}
